Sources/NIOWebSocket/WebSocketFrameDecoder.swift

```diff
@@ -0,0 +1,321 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebSocketFrameDecoder.swift
+//  NIOWebSocket
+//
+//  Created by Cory Benfield on 15/02/2018.
+//
+
+import NIO
+
+/// Errors thrown by the NIO websocket module.
+public enum NIOWebSocketError: Error {
+    /// The frame being sent is larger than the configured maximum
+    /// acceptable frame size
+    case invalidFrameLength
+
+    /// A control frame may not be fragmented.
+    case fragmentedControlFrame
+
+    /// A control frame may not have a length more than 125 bytes.
+    case multiByteControlFrameLength
+}
+
+internal extension WebSocketErrorCode {
+    internal init(_ error: NIOWebSocketError) {
+        switch error {
+        case .invalidFrameLength:
+            self = .messageTooLarge
+        case .fragmentedControlFrame,
+             .multiByteControlFrameLength:
+            self = .protocolError
+        }
+    }
+}
+
+public extension ByteBuffer {
+    /// Applies the WebSocket unmasking operation.
+    ///
+    /// - parameters:
+    ///     - maskingKey: The masking key.
+    public mutating func unmask(_ maskingKey: WebSocketMaskingKey, indexOffset: Int = 0) {
```

Comment:
this is publicly available so it should have something with websockets in the name.